src/main/java/frc/robot/subsystems/Vision/VisionIO.java

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems.vision;
+
+import edu.wpi.first.math.geometry.Pose3d;
+import java.util.ArrayList;
+import java.util.List;
+import org.photonvision.targeting.PhotonTrackedTarget;
+
+/** Add your docs here. */
+public interface VisionIO {
+
+  public static class VisionIOInputs {
+    public double timestamp = 0.0;
+    public double latency = 0.0;
+    public List<PhotonTrackedTarget> targets =
+        new ArrayList<>(); // TODO make protobuf work whenever that happens
+    public double numTags = 0;
+    public Pose3d pose = new Pose3d();
+  }
+
+  public default void updateInputs(VisionIOInputs inputs, Pose3d pose) {};
```

Comment:
do you always need the pose or just in sim?

Reply:
i think just sim, but i still need to figure that out

Reply:
if its just sim maybe make the sim io constructor take in a `Supplier<Pose3d>` which it calls instead of having `updateInputs` always take it

Reply:
but then how do i get it to update every cycle? (sorry if this is really obvious and i'm just missing something)